java/rocksjni/rocksjni.cc

```diff
@@ -1587,3 +1587,74 @@ jlong Java_org_rocksdb_RocksDB_getUpdatesSince(JNIEnv* env,
   rocksdb::RocksDBExceptionJni::ThrowNew(env, s);
   return 0;
 }
+
+//////////////////////////////////////////////////////////////////////////////
+// rocksdb::DB::AddFile
+
+void add_file_helper(JNIEnv* env, jobjectArray& jfile_path_list,
+                     std::vector<std::string>& file_path_list) {
+  const int jfile_path_list_len =
+      static_cast<int>(env->GetArrayLength(jfile_path_list));
+  for (int i = 0; i < jfile_path_list_len; i++) {
+    jstring jfile_path =
+        static_cast<jstring>(env->GetObjectArrayElement(jfile_path_list, i));
+    const char* file_path = env->GetStringUTFChars(jfile_path, NULL);
+    file_path_list.push_back(std::string(file_path));
+    env->ReleaseStringUTFChars(jfile_path, file_path);
+  }
```

Comment:
Within the loop, I might suggest that you should use `DeleteLocalRef` to cleanup the reference created by `GetObjectArrayElement`. Otherwise you will run out of local references if you have a large array.